packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -1,31 +1,164 @@
+import ConfirmBulkActionDialog from '../ConfirmBulkActionDialog';
+
+const confirmDialogsPropTypes = {
+  isConfirmButtonLoading: PropTypes.bool.isRequired,
+  isOpen: PropTypes.bool.isRequired,
+  onConfirm: PropTypes.func.isRequired,
+  onToggleDialog: PropTypes.func.isRequired,
+};
+
+const ConfirmDialogPublishAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <ConfirmBulkActionDialog
+      isOpen={isOpen}
+      onToggleDialog={onToggleDialog}
+      icon={<ExclamationMarkCircle />}
+      dialogBody={
+        <Flex direction="column" alignItems="stretch" gap={2}>
+          <Typography id="confirm-description" textAlign="center">
+            {formatMessage({
+              id: getTrad('popUpWarning.bodyMessage.contentType.publish.all'),
+              defaultMessage: 'Are you sure you want to publish these entries?',
+            })}
+          </Typography>
+          <InjectionZoneList area="contentManager.listView.publishModalAdditionalInfos" />
+        </Flex>
+      }
+      endAction={
+        <Button
+          onClick={onConfirm}
+          variant="secondary"
+          startIcon={<Check />}
+          loading={isConfirmButtonLoading}
+        >
+          {formatMessage({
+            id: 'app.utils.publish',
+            defaultMessage: 'Publish',
+          })}
+        </Button>
+      }
+    />
+  );
+};
+
+ConfirmDialogPublishAll.propTypes = confirmDialogsPropTypes;
+
+const ConfirmDialogUnpublishAll = ({
+  isOpen,
+  onToggleDialog,
+  isConfirmButtonLoading,
+  onConfirm,
+}) => {
+  const { formatMessage } = useIntl();
 
-import ConfirmDialogDeleteAll from '../ConfirmDialogDeleteAll';
+  return (
+    <ConfirmBulkActionDialog
+      isOpen={isOpen}
+      onToggleDialog={onToggleDialog}
+      icon={<ExclamationMarkCircle />}
+      dialogBody={
+        <Flex direction="column" alignItems="stretch" gap={2}>
+          <Typography id="confirm-description" textAlign="center">
+            {formatMessage({
+              id: getTrad('popUpWarning.bodyMessage.contentType.unpublish.all'),
+              defaultMessage: 'Are you sure you want to unpublish these entries?',
+            })}
+          </Typography>
+          <Flex>
+            <InjectionZoneList area="contentManager.listView.unpublishModalAdditionalInfos" />
+          </Flex>
+        </Flex>
+      }
+      endAction={
+        <Button
+          onClick={onConfirm}
+          variant="secondary"
+          startIcon={<Check />}
+          loading={isConfirmButtonLoading}
+        >
+          {formatMessage({
+            id: 'app.utils.unpublish',
+            defaultMessage: 'Unpublish',
+          })}
+        </Button>
+      }
+    />
+  );
+};
+
+ConfirmDialogUnpublishAll.propTypes = confirmDialogsPropTypes;
+
+const ConfirmDialogDeleteAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
```

Comment:
I'm don't know if it matter or not but delete looks different with i18n. I think consisten formatting could be nice and also consistent wording (omit only):

![Screenshot 2023-05-03 at 10 17 48](https://user-images.githubusercontent.com/26598053/235864621-9c8b0276-3386-4896-9d9e-277c484ab973.png)

![Screenshot 2023-05-03 at 10 17 34](https://user-images.githubusercontent.com/26598053/235864577-492e0008-4077-4591-b437-7da4a3878885.png)